Mathlib/Data/Fin/Basic.lean

```diff
@@ -777,6 +777,17 @@ def castLEEmb (h : n ≤ m) : Fin n ↪o Fin m :=
 #align fin.cast_le_mk Fin.castLE_mk
 #align fin.cast_le_zero Fin.castLE_zero
 
+@[simp] lemma castLE_castSucc {n m} (i : Fin n) (h : (n+1) ≤ m) :
+    i.castSucc.castLE h = i.castLE (Nat.le_of_succ_le h) :=
+  rfl
+
+@[simp] lemma castLE_comp_castSucc {n m} (h : (n+1) ≤ m) :
+    (Fin.castLE h ∘ Fin.castSucc) = Fin.castLE (Nat.le_of_succ_le h) :=
+  rfl
+
+@[simp] lemma castLE_refl (n : ℕ) : Fin.castLE (Nat.le_refl n) = id :=
+  rfl
```

Comment:
These simp lemmas are no longer needed for the other proofs in this PR, but they still seem like sensible simp lemmas nonetheless. Should we keep them in, or should I remove them from this PR?

Reply:
These look like good lemmas and are ok to keep here, but should be mentioned in the PR description.